trieste/acquisition/rule.py

```diff
@@ -702,7 +705,10 @@ def state_func(
         return state_func
 
 
-class DiscreteThompsonSampling(AcquisitionRule[TensorType, SearchSpace, ProbabilisticModel]):
+T = TypeVar("T", bound=ProbabilisticModel, contravariant=True)
```

Comment:
Please could you explain this to me

Reply:
This is similar to your previous question (but a bit simpler): DiscreteThompsonSampling can support any model, as long as the thomspon sampler supports it. The default sampler (ExactThompsonSampler) supports all ProbabilisticModels, but ThompsonSamplerFromTrajectory for example requires the modles to provide a trajectory sampler.